### PR TITLE
Confirm runnning Pools before restarting existing pools

### PR DESCRIPTION
**Pull request by dvaldivia**
This makes sure we can reach the first pod of a pool before telling other pools to restart to account for the new pool, this is to avoid a deadlock  where minio becomes non-functional and the statefulsets are never updated.

Also makes it so we create the log search secret before creating ss since we may reference them

Signed-off-by: Daniel Valdivia <18384552+dvaldivia@users.noreply.github.com>

### CODE REVIEW

**Review by Alevsk — APPROVED** [review]

File: pkg/controller/cluster/monitoring.go [hunk 1]
```diff
@@ -77,52 +77,56 @@ func (c *Controller) tenantsHealthMonitor() error {
...
-		adminClnt, err := tenant.NewMinIOAdmin(minioSecret.Data)
+		adminClnt, err := tenant.NewMinIOAdmin(tenantConfiguration)
 		if err != nil {
 			// show the error and continue
 			klog.V(2).Infof(err.Error())
 			continue
 		}
-
-		srvInfoCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		klog.Infof("dos")
```
Comment (bexsoft) [inline pkg/controller/cluster/monitoring.go:102]: Please remove this

File: pkg/controller/cluster/monitoring.go [hunk 2]
```diff
@@ -77,52 +77,56 @@ func (c *Controller) tenantsHealthMonitor() error {
...
 		defer cancel()
 		storageInfo, err := adminClnt.StorageInfo(srvInfoCtx)
 		if err != nil {
 			// show the error and continue
-			klog.V(2).Infof(err.Error())
+			klog.Infof("'%s/%s' Failed to get storage info: %v", tenant.Namespace, tenant.Name, err)
 			continue
 		}
 
+		klog.Infof("dos.5 %+v", storageInfo)
```
Comment (bexsoft) [inline pkg/controller/cluster/monitoring.go:112]: Please remove this

File: pkg/controller/cluster/monitoring.go [hunk 3]
```diff
@@ -137,11 +141,14 @@ func (c *Controller) tenantsHealthMonitor() error {
 		if tenant.Status.DrivesOnline < tenant.Status.WriteQuorum {
 			tenant.Status.HealthStatus = miniov2.HealthStatusRed
 		}
+		klog.Infof("trres")
```
Comment (bexsoft) [inline pkg/controller/cluster/monitoring.go:144]: Please remove this

File: pkg/controller/cluster/monitoring.go [hunk 4]
```diff
@@ -137,11 +141,14 @@ func (c *Controller) tenantsHealthMonitor() error {
...
 
-		if _, err = c.updatePoolStatus(context.Background(), tenant); err != nil {
-			klog.V(2).Infof(err.Error())
+		if _, err = c.updatePoolStatus(context.Background(), &tenant); err != nil {
+			klog.Infof("'%s/%s' Can't update tenant status: %v", tenant.Namespace, tenant.Name, err)
 		}
 
+		klog.Infof("safe")
```
Comment (bexsoft) [inline pkg/controller/cluster/monitoring.go:150]: Please remove this

**Review by bexsoft — CHANGES_REQUESTED** [review]

File: pkg/controller/cluster/monitoring.go (hunk 1 above)
Comment (dvaldivia) [reply pkg/controller/cluster/monitoring.go:102]: removed

File: pkg/controller/cluster/monitoring.go (hunk 2 above)
Comment (dvaldivia) [reply pkg/controller/cluster/monitoring.go:112]: removed

File: pkg/controller/cluster/monitoring.go (hunk 3 above)
Comment (dvaldivia) [reply pkg/controller/cluster/monitoring.go:144]: removed

File: pkg/controller/cluster/monitoring.go (hunk 4 above)
Comment (dvaldivia) [reply pkg/controller/cluster/monitoring.go:150]: removed

**Review by bexsoft — APPROVED** [review]

**Review by Alevsk — APPROVED** [review]

**Pull request merged**